### ISSUE TITLE
Remove `iterator` and `iterator/iteratortest` packages

### ISSUE THREAD

**Issue by jbduncan** [jbduncan/go-containers#20]
Blocked on the currently-experimental [range-over-func](https://github.com/golang/go/issues/61405) Go feature being merged into a stable version of Go.

When range-over-func is released, classic-style iterators are no longer needed, so they can be removed.

Note: consider porting `iterator/iteratortest` to https://github.com/jbduncan/jupiter-collection-testers or JUnit Pioneer first.

**Comment by jbduncan** [comment]
Done.
